lib/metasploit/framework/login_scanner/softing_sis.rb

```diff
@@ -34,17 +34,11 @@ def check_setup
           false
         end
 
-        # the actual login method, called by #attempt_login
+        # get the authentication token
         #
-        # @param user [String] The username to try
-        # @param pass [String] The password to try
-        # @return [Hash]
-        #   * status [Metasploit::Model::Login::Status]
-        #   * proof [String] the HTTP response body
-        def do_login(user, pass)
-          # prep the data needed for login
-          protocol = ssl ? 'https' : 'http'
-          # attempt to get an authentication token
+        # @param user [String] The username
+        # @return [String] The authentication token
```

Comment:
This method should be adjusted to always return the same data type. According to the doc string here, the return type is the authentication token as a string, however there are multiple instances where if the method fails a hash is actually returned.

Since the status and proof values are necessary, I would recommend changing the function to *always* return a hash, that *always* has a status key. Then the caller would receive the hash, and check the status before either extracting the auth token itself, or doing whatever it needs to with the proof. On success, set `status` to `LOGIN_STATUS::SUCCESSFUL`.

In either case, it should always return the same data type (or nil) and the documentation should be accurate.

Reply:
Should be fixed with 402614180983bd514ce57f6fa6297ea0347538b8